app/models/manageiq/providers/vmware/cloud_manager/vm/operations/power.rb

```diff
@@ -0,0 +1,26 @@
+module ManageIQ::Providers::Vmware::CloudManager::Vm::Operations::Power
+  def validate_pause
+    validate_unsupported("Pause operation")
+  end
+
+  def raw_start
+    vm.power_on
+  end
+
+  def raw_stop
+    vm.power_off
+  end
+
+  def raw_suspend
+    vm.suspend
+  end
+
+  def raw_restart
+    vm.reset
+  end
+
+  def vm
+    vcd = ext_management_system.connect
+    vm  = vcd.vms.get_single_vm(uid_ems)
+  end
```

Comment:
If you add a `#provider_object` method like [here](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/google/cloud_manager/vm.rb#L4l) you can say `with_provider_object(&:reset)` which is a little more common in the rest of the app ([example](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/google/cloud_manager/vm/operations/power.rb#L26))